### PR TITLE
B2b cart comments smoke test

### PR DESCRIPTION
**Pull request by durgesh-verma-spryker**
## PR Description
https://spryker.atlassian.net/browse/CC-21651

Add comments to a Cart

- As a buyer user add few comments to a Cart in Yves side, place an order and check comments appears at Order detailed page in Zed and Order detailed page in Yves

- At Yves Add, Edit, Delete comments in Cart

### CODE REVIEW

File: tests/smoke/smoke_b2b.robot [hunk 1]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: the following shopping cart is shown:    shoppingCartName+${random}    Owner access
+    Yves: go to PDP of the product with sku:    ${bundled_product_3_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to PDP of the product with sku:    ${bundled_product_2_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to the shopping cart through the header with name:    shoppingCartName+${random}
+    Yves: 'Shopping Cart' page is displayed
+    Yves: shopping cart contains the following products:    ${bundled_product_2_abstract_sku}
+    Yves: go to 'Customer Account' page
+    Yves: go to user menu item in the left bar:    Order History
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:987]: this test looks not full. Where are assertions in Zed? 

Apply same comments from previous test, like name of the test, name of the shopping cart, 2 products, additional not needed check of the cart and item in cart

**Review by AntonShopin — CHANGES_REQUESTED** [review]
I also don't see tests for:
* View and manage comments in the order history (Yves)

File: resources/pages/yves/yves_order_details_page.robot [hunk 2]
```diff
@@ -2,4 +2,12 @@
 ${order_details_main_content_locator}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']
 ${order_details_reorder_all_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder all')]
 ${order_details_reorder_selected_items_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder selected items')]
-${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
\ No newline at end of file
+${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
+${order_details_write_comment_placeholder}    xpath=//textarea[@placeholder='Write a comment']
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: Fixed

File: resources/pages/yves/yves_order_details_page.robot [hunk 3]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_edit_comment_placeholder}    xpath=//textarea[@required="required"]
+${order_details_add_button}    xpath=//button[@title='Add']
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: Fixed

File: resources/pages/yves/yves_order_details_page.robot [hunk 4]
```diff
@@ -2,4 +2,12 @@
 ${order_details_main_content_locator}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']
 ${order_details_reorder_all_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder all')]
 ${order_details_reorder_selected_items_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder selected items')]
-${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
\ No newline at end of file
+${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
+${order_details_write_comment_placeholder}    xpath=//textarea[@placeholder='Write a comment']
+${order_details_edit_comment_placeholder}    xpath=//textarea[@required="required"]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: Fixed

File: resources/pages/yves/yves_order_details_page.robot [hunk 5]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_add_button}    xpath=//button[@title='Add']
+${order_details_edit_comment_button}    xpath=(//div[@class="grid grid--center"]//button)[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: FIxed

File: resources/pages/yves/yves_order_details_page.robot [hunk 6]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_update_button}    xpath=//button[@class="button button--small js-comment-form__submit-button"]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: Fixed

File: resources/pages/yves/yves_order_details_page.robot [hunk 7]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_remove_comment_button}    xpath=//button[@action="/en/comment/remove"]
+${yves_order_details_page_comments}    xpath=//div[contains(@class,'comment-form__readonly')]//p
+${yves_comment_delete_button}    xpath=(//button[contains(@action,'/comment/remove')])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot]: Fixed

File: resources/pages/zed/zed_order_details_page.robot [hunk 8]
```diff
@@ -1 +1,4 @@
 *** Variables ***
+${zed_order_details_page_comments}    xpath=//p[@class="comment-title"]//following-sibling::p
+${order_details_page_add_comments_textbox}    xpath=//textarea[@class="textarea textarea--cart-comment add-comment-form__textarea js-add-comment-form__textarea"]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_order_details_page.robot]: Fixed

File: resources/steps/comments_order_steps.robot [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+Resource    ../common/common_yves.robot
+Resource    ../common/common.robot
+Resource    ../steps/customer_account_steps.robot
+Resource    ../pages/zed/zed_order_details_page.robot
+Resource    ../pages/yves/yves_order_details_page.robot
+Resource    ../common/common_zed.robot
+
+*** Variables ***
+${view_button_of_last_order}    xpath=(//a[contains(@href,'/customer/order/details')])[1]
+${zed_view_button_of_last_order}    xpath=(//a[contains(@href,'/sales/detail')])[1]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/comments_order_steps.robot]: Fixed

File: resources/steps/comments_order_steps.robot [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+*** Keywords ***
+Yves: adding comments on cart before checkout
+    Click    ${order_details_write_comment_placeholder}
+    Type Text    ${order_details_write_comment_placeholder}    abc${random}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/comments_order_steps.robot]: Fixed

File: resources/steps/comments_order_steps.robot [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+Zed: checking comments added in zed
+    Click    ${zed_view_button_of_last_order}
+    Wait Until Element Is Visible    ${zed_order_details_page_comments}
+    ${text_entered}    Get Text    ${zed_order_details_page_comments}
+       IF    '${text_entered}' == 'abc${random}'
+            Log    comments added to cart appears in order details page in zed
+       ELSE
+            Log    comments added to cart does not appears in order details page in zed
+       END
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/comments_order_steps.robot]: Fixed

File: resources/steps/comments_order_steps.robot [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+Yves: add edit delete comments on cart
+    Type Text    ${order_details_write_comment_placeholder}    abc${random}
+    Click    ${order_details_add_button}
+    Click    ${order_details_edit_comment_button}
+    Click    ${order_details_edit_comment_placeholder}
+    Clear Text    ${order_details_edit_comment_placeholder}  
+    Keyboard Key    press    Tab
+    Input Text    ${order_details_edit_comment_placeholder}    xyz${random}
+    Click    ${order_details_update_button}
+    Click    ${order_details_remove_comment_button}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/comments_order_steps.robot]: Fixed

File: resources/steps/comments_order_steps.robot [hunk 13]
```diff
@@ -0,0 +1,49 @@
+*** Settings ***
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/comments_order_steps.robot]: Fixed

File: resources/steps/zed_order_shipment_steps.robot [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+        IF    '${key}'=='lname'    Type Text    ${shipment_address_last_name}    ${value}
+        IF    '${key}'=='email'    Type Text    ${shipment_address_email}    ${value}
+        IF    '${key}'=='Address1'    Type Text    ${shipment_address_address1}     ${value}
+        IF    '${key}'=='address2'    Type Text    ${shipment_address_address2}     ${value}
+        IF    '${key}'=='city'    Type Text    ${shipment_address_city}     ${value}
+        IF    '${key}'=='zipcode'    Type Text    ${shipment_address_zipcode}     ${value}
+    END
+    Select From List By Value    ${salutation_create_new_shipment_page}    Mr
+    Select From List By Index    ${country_create_new_shipment_page}    4    
+    Select From List By Text    ${shipment_method_dropdown_new_shipment_page}    DHL - Express
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/zed_order_shipment_steps.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 15]
```diff
@@ -24,6 +24,8 @@ Resource    ../../resources/steps/zed_customer_steps.robot
 Resource    ../../resources/steps/zed_discount_steps.robot
 Resource    ../../resources/steps/zed_availability_steps.robot
 Resource    ../../resources/steps/zed_cms_page_steps.robot
+Resource    ../../resources/steps/zed_order_shipment_steps.robot
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 16]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: click on the 'Checkout' button in the shopping cart
+    Yves: billing address same as shipping address:    true
+    Yves: select the following existing address on the checkout as 'shipping' address and go next:    ${yves_company_user_shared_permission_owner_address}
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    [Teardown]    Run Keywords    Yves: logout on Yves as a customer
+    ...    AND    Reload
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 17]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: create shipment 
+    Zed: flash message should be shown:    success
+
+Edit_shipment_in_zed
+    [Documentation]    editing shippment on order in zed
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: edit shipment
+    Zed: flash message should be shown:    success
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 18]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+Add_comment_on_carts
+    [Documentation]    adding comments to cart
+    Yves: login on Yves with provided credentials:    ${yves_company_user_shared_permission_owner_email}
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: create new 'Shopping Cart' with name:    shoppingCartName+${random}
+    Yves: 'Shopping Carts' widget contains:    shoppingCartName+${random}    Owner access
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: the following shopping cart is shown:    shoppingCartName+${random}    Owner access
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 19]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: go to PDP of the product with sku:    ${bundled_product_3_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to PDP of the product with sku:    ${bundled_product_2_abstract_sku
+    Yves: add product to the shopping cart
+    Yves: go to the shopping cart through the header with name:    shoppingCartName+${random}
+    Yves: 'Shopping Cart' page is displayed
+    Yves: shopping cart contains the following products:    ${bundled_product_3_concrete_sku}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:879]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 20]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: edit shipment
+    Zed: flash message should be shown:    success
+
+Add_comment_on_carts
+    [Documentation]    adding comments to cart
+    Yves: login on Yves with provided credentials:    ${yves_company_user_shared_permission_owner_email}
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: create new 'Shopping Cart' with name:    shoppingCartName+${random}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 21]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: 'Shopping Carts' widget contains:    shoppingCartName+${random}    Owner access
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: the following shopping cart is shown:    shoppingCartName+${random}    Owner access
+    Yves: go to PDP of the product with sku:    ${bundled_product_3_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to PDP of the product with sku:    ${bundled_product_2_abstract_sku
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 22]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: add product to the shopping cart
+    Yves: go to the shopping cart through the header with name:    shoppingCartName+${random}
+    Yves: 'Shopping Cart' page is displayed
+    Yves: shopping cart contains the following products:    ${bundled_product_3_concrete_sku}
+    Yves: adding comments on cart before checkout
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 23]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: click on the 'Checkout' button in the shopping cart
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:881]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 24]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: select the following existing address on the checkout as 'shipping' address and go next:    ${yves_company_user_shared_permission_owner_address}
+    Yves: select the following shipping method on the checkout and go next:    Express
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    Yves: go to 'Order History' page to check comments
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: checking comments added in zed
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 25]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    [Teardown]    Run Keywords    Yves: logout on Yves as a customer
+    ...    AND    Reload
+
+At_Yves_Add_Edit_Delete_comments_in_Cart
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 26]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    [Documentation]    adding editing and deleting comments in carts
+    Yves: login on Yves with provided credentials:    ${yves_company_user_shared_permission_owner_email}
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: create new 'Shopping Cart' with name:    shoppingCartName+${random}
+    Yves: 'Shopping Carts' widget contains:    shoppingCartName+${random}    Owner access
+    Yves: go to 'Shopping Carts' page through the header
+    Yves: 'Shopping Carts' page is displayed
+    Yves: the following shopping cart is shown:    shoppingCartName+${random}    Owner access
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 27]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: go to PDP of the product with sku:    ${bundled_product_3_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to PDP of the product with sku:    ${bundled_product_2_abstract_sku}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 28]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: add product to the shopping cart
+    Yves: go to the shopping cart through the header with name:    shoppingCartName+${random}
+    Yves: 'Shopping Cart' page is displayed
+    Yves: shopping cart contains the following products:    ${bundled_product_2_abstract_sku}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 29]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: add edit delete comments on cart
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot [hunk 30]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    [Teardown]    Run KeywordS    Yves: delete 'Shopping Cart' with name:    shoppingCartName+${random}
+    ...    AND     Yves: logout on Yves as a customer
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 1 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: resources/steps/order_comments_steps.robot [hunk 31]
```diff
@@ -0,0 +1,69 @@
...
+*** Keywords ***
+Yves: adding comments on cart before checkout:
+    [Arguments]    ${comments}
+    Click    ${order_details_write_comment_placeholder}
+    Type Text    ${order_details_write_comment_placeholder}    ${comments}
+    Click    ${order_details_add_button}
+    ${text}    Get Text    ${comments_shopping_cart}
+    Should Be Equal    ${comments}    ${text}
+
+Yves: go to 'Order History' page to check comments:
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:18]: please rename it. Order History page is a page where you see the list of all orders in a table. In your keyword, you go order details page of the last place order

File: resources/steps/order_comments_steps.robot [hunk 32]
```diff
@@ -0,0 +1,69 @@
...
+            Log    comments added to cart appears in order details page in zed
+       ELSE
+            Fail    comments added to cart does not appears in order details page in zed
+       END
+
+Yves: add comments on cart:
+    [Arguments]    ${comment}
+    Type Text    ${order_details_write_comment_placeholder}    ${comment}
+    Click    ${order_details_add_button}
+    ${text}    Get Text    ${comments_shopping_cart}
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:45]: What is the difference between this keyword and `Yves: adding comments on cart before checkout:`?

File: resources/steps/order_comments_steps.robot [hunk 33]
```diff
@@ -0,0 +1,69 @@
...
+    Should Be Equal    ${comment}    ${text}
+
+Yves:edit comments on cart:
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:48]: 1. Add a space after `:`
2. Why do you put everywhere 'comments' in plural? We can add multiple comments with your keywords?

File: resources/steps/order_comments_steps.robot [hunk 34]
```diff
@@ -0,0 +1,69 @@
...
+    [Arguments]    ${comment}
+    Click    ${order_details_edit_comment_button}
+    Click    ${order_details_edit_comment_placeholder}
+    Clear Text    ${order_details_edit_comment_placeholder}  
+    Keyboard Key    press    Tab
+    Input Text    ${order_details_edit_comment_placeholder}    ${comment}
+    Click    ${order_details_update_button}
+    ${text}    Get Text    ${comments_shopping_cart}
+    Should Be Equal    ${comment}    ${text}
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:57]: as above, this assertion should not be here

File: resources/steps/order_comments_steps.robot [hunk 35]
```diff
@@ -0,0 +1,69 @@
...
+
+Yves: delete comments on cart
+    Click    ${order_details_remove_comment_button} 
+    Page Should Not Contain Element    ${order_details_remove_comment_button}
+
+Yves: add comment on order:
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:63]: please rename it so it's clear where we add this comment.

File: resources/steps/order_comments_steps.robot [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+    [Arguments]    ${comments}
+    Click    ${order_details_page_add_comments_textbox}
+    Type Text    ${order_details_page_add_comments_textbox}    ${comment}
+    Click    ${add_button_order_details_page}
+    ${text}    Get Text    ${comments_shopping_cart}
+    Should Be Equal    ${comments}    ${text}
```
Comment (AntonShopin) [inline resources/steps/order_comments_steps.robot:69]: again, those assertions should not be here

File: tests/smoke/smoke_b2b.robot (hunk 19 above)
[the remark this answers is not in the document]
Comment (AntonShopin) [reply tests/smoke/smoke_b2b.robot:879]: not fixed. I asked you to remove this step

File: tests/smoke/smoke_b2b.robot (hunk 23 above)
[the remark this answers is not in the document]
Comment (AntonShopin) [reply tests/smoke/smoke_b2b.robot:881]: not fixed. See my comments above about your keywords, where you have injected assertion. It should be a separate keyword

File: tests/smoke/smoke_b2b.robot [hunk 37]
```diff
@@ -866,4 +867,63 @@ Refunds
...
+Comments_Management_in_the_Cart
+    [Documentation]    adding editing and deleting comments in carts
+    Yves: login on Yves with provided credentials:    ${yves_company_user_shared_permission_owner_email}
+    Yves: create new 'Shopping Cart' with name:    comments+${random}
+    Yves: go to PDP of the product with sku:    ${bundled_product_3_abstract_sku}
+    Yves: add product to the shopping cart
+    Yves: go to the shopping cart through the header with name:    comments+${random}
+    Yves: add comments on cart:    abc${random}
+    Yves: edit comments on cart:    xyz${random}
+    Yves: delete comments on cart
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:904]: How do you assert that a specific comment is removed? 
Cart can contain few comments, please think one more time about how to handle it in all your keywords

File: tests/smoke/smoke_b2b.robot [hunk 38]
```diff
@@ -866,4 +867,63 @@ Refunds
...
+    [Teardown]    Run Keyword    Yves: delete 'Shopping Cart' with name:    comments+${random}
+        
+At_Yves_attach_few_comments_to_order_check_they_appeared_in_Zed
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2b.robot:907]: Test name should be changed

File: tests/smoke/smoke_b2b.robot [hunk 39]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: 'Thank you' page is displayed
+    Yves: go to 'Order History' page to check comments
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: checking comments added in zed
+    [Teardown]    Run Keywords    Yves: logout on Yves as a customer
+    ...    AND    Reload
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 23 above)
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:881]: Fixed, added new keyword

File: resources/pages/yves/yves_order_details_page.robot [hunk 40]
```diff
@@ -2,4 +2,12 @@
 ${order_details_main_content_locator}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']
 ${order_details_reorder_all_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder all')]
 ${order_details_reorder_selected_items_button}    xpath=//customer-reorder-form[@data-qa='component customer-reorder-form']//button[contains(.,'Reorder selected items')]
-${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
\ No newline at end of file
+${order_details_shipping_address_locator}    xpath=//div[@class='summary-sidebar__item']//ul[@data-qa='component display-address']
+${order_details_write_comment_placeholder}    xpath=//div[@data-qa='component add-comment-form']//form[@method='POST']/textarea
+${order_details_edit_comment_placeholder}    xpath=//div[starts-with(@class,"comment-form__edit js-comment__target-all")]/textarea
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot:7]: Fixed, as per your suggestion

File: tests/smoke/smoke_b2b.robot [hunk 41]
```diff
@@ -866,4 +868,120 @@ Refunds
...
+
+Edit_shipment_in_zed
+    [Documentation]    editing shippment on order in zed
+    Zed: login on Zed with provided credentials:    ${zed_admin_email}
+    Zed: go to second navigation item level:    Sales    Orders
+    Zed: edit shipment
+    Zed: flash message should be shown:    success
+
+Add_comment_on_carts
+    [Documentation]    adding comments to cart
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot]: Fixed, added documentation

File: tests/smoke/smoke_b2b.robot (hunk 19 above)
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:879]: Removed successfully

File: resources/pages/yves/yves_order_details_page.robot [hunk 42]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_add_button}    xpath=//button[@class="button button--primary add-comment-form__button js-add-comment-form__button"]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot:8]: Fixed, name and locators updated

File: resources/pages/yves/yves_order_details_page.robot [hunk 43]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_edit_comment_button}    xpath=//button[contains(@class,"col button button--hollow-icon-small js-comment__trigger-all")]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot:9]: Fixed

File: resources/pages/yves/yves_order_details_page.robot [hunk 44]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_update_button}    xpath=//button[contains(@action,"/en/comment/update")]
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot:10]: Fixed, as suggested

File: resources/pages/yves/yves_order_details_page.robot [hunk 45]
```diff
@@ -2,4 +2,12 @@
...
+${order_details_remove_comment_button}    xpath=//button[contains(@action,"/en/comment/remove")]
+${yves_order_details_page_comments}    xpath=//div[contains(@class,'comment-form__readonly')]//p
+${comments_shopping_cart}    xpath=//div[contains(@class,"comment-form__readonly js-comment__target-all")]/p
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/yves/yves_order_details_page.robot:13]: Fixed, added locators to respective page

File: resources/pages/zed/zed_order_details_page.robot [hunk 46]
```diff
@@ -1 +1,4 @@
 *** Variables ***
+${zed_order_details_page_comments}    xpath=//p[@class="comment-title"]//following-sibling::p
+${order_details_page_add_comments_textbox}    xpath=//form[contains(@action,"/comment/add")]/textarea
+${add_button_order_details_page}    xpath=//form[contains(@action,"/comment/add")]/button
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/pages/zed/zed_order_details_page.robot:4]: Fixed

File: resources/steps/order_comments_steps.robot [hunk 47]
```diff
@@ -0,0 +1,69 @@
+*** Settings ***
+Resource    ../common/common_yves.robot
+Resource    ../common/common.robot
+Resource    ../steps/customer_account_steps.robot
+Resource    ../pages/zed/zed_order_details_page.robot
+Resource    ../pages/yves/yves_order_details_page.robot
+Resource    ../common/common_zed.robot
+Resource    order_history_steps.robot
+*** Keywords ***
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:9]: Fixed, Added .

File: resources/steps/order_comments_steps.robot [hunk 48]
```diff
@@ -0,0 +1,69 @@
...
+Yves: adding comments on cart before checkout:
+    [Arguments]    ${comments}
+    Click    ${order_details_write_comment_placeholder}
+    Type Text    ${order_details_write_comment_placeholder}    ${comments}
+    Click    ${order_details_add_button}
+    ${text}    Get Text    ${comments_shopping_cart}
+    Should Be Equal    ${comments}    ${text}
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:16]: Fixed, added assertion

File: resources/steps/order_comments_steps.robot (hunk 31 above)
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:18]: Fixed

File: resources/steps/order_comments_steps.robot [hunk 49]
```diff
@@ -0,0 +1,69 @@
...
+    Yves: go to 'Order History' page
+    Yves: 'View Order/Reorder/Return' on the order history page:    View Order    ${lastPlacedOrder}
+    ${text_entered}    Get Text    ${yves_order_details_page_comments}
+    IF    '${text_entered}' == '${comments}'
+        Log    text entered in comments appers in order details page in Yves
+    ELSE
+        Fail    comments not added succesfully to order details page in Yves
+    END    
+
+Zed: checking comments added in zed:
```
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:29]: Fixed

File: resources/steps/order_comments_steps.robot (hunk 32 above)
[the remark this answers is not in the document]
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:45]: Fixed

Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:45]: Fixed, removed duplicate successfully

File: resources/steps/order_comments_steps.robot (hunk 33 above)
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:48]: Fixed

File: resources/steps/order_comments_steps.robot (hunk 34 above)
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:57]: Fixed, Added new keyword for assertion

File: resources/steps/order_comments_steps.robot (hunk 35 above)
Comment (durgesh-verma-spryker) [reply resources/steps/order_comments_steps.robot:63]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 37 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:904]: Fixed

File: tests/smoke/smoke_b2b.robot (hunk 38 above)
Comment (durgesh-verma-spryker) [reply tests/smoke/smoke_b2b.robot:907]: Fixed

**Pull request closed without merging**